packages/@lwc/engine-core/src/framework/renderer.ts

```diff
@@ -46,6 +46,13 @@ export interface Renderer<N = HostNode, E = HostElement> {
     isConnected(node: N): boolean;
     insertGlobalStylesheet(content: string): void;
     assertInstanceOfHTMLElement?(elm: any, msg: string): void;
+    defineCustomElement(
```

Comment:
3 new APIs added to renderer to be able to accomodate the registration of the element.